bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/logic/CommandTag.java

```diff
@@ -0,0 +1,181 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.icalendar.internal.logic;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.regex.PatternSyntaxException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.HSBType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.OpenClosedType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.library.types.PlayPauseType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.types.RewindFastforwardType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.types.UpDownType;
+import org.eclipse.smarthome.core.types.Command;
+
+/**
+ *
+ * @author Andrew Fiddian-Green - Initial contribution
+ *
+ *         This is a class that implements a Command Tag that may be embedded in an Event Description
+ *
+ *         Valid Tags must follow one of the following forms..
+ *
+ *         BEGIN:<itemName>:<targetState>
+ *         BEGIN:<itemName>:<targetState>:<authorizationCode>
+ *         END:<itemName>:<targetState>
+ *         END:<itemName>:<targetState>:<authorizationCode>
+ *
+ */
+@NonNullByDefault
+public class CommandTag {
+    @Nullable
+    public String itemName;
+
+    @Nullable
+    public String targetState;
+
+    @Nullable
+    public String fullTag;
+
+    @Nullable
+    public CommandTagType tagType;
+
+    protected boolean isValid = false;
+
+    @Nullable
+    private String authorizationCode;
+
+    private @Nullable Command castToCommandType(Class<? extends Command> commandType) {
+        try {
+            Method valueOf = commandType.getMethod("valueOf", String.class);
+            Command cmd = (Command) valueOf.invoke(commandType, targetState);
+            if (cmd != null) {
+                return cmd;
+            }
+        } catch (NoSuchMethodException e) {
+        } catch (IllegalArgumentException e) {
+        } catch (IllegalAccessException e) {
+        } catch (InvocationTargetException e) {
+        }
+        return null;
+    }
+
+    public CommandTag(String line) {
+        String[] fields;
+        try {
```

Comment:
This try block won't ever trigger.

Reply:
I don't know what you are referring to here. The String.split() method can certainly throw an exception (it says  @throws  PatternSyntaxException if the regular expression's syntax is invalid)..

Reply:
I am trying to catch any error that might happen when line.split(“:”) fails for whatever reason; as you say it is not a problem with the input pattern “:” but likely with the source text “line”. If this is not PatterSyntaxException, then please tell me what it is.

Reply:
PatternSyntaxException can never come from the source string.

Reply:
Resolved